stdlib/test/builtin/test_file.mojo

```diff
@@ -160,6 +160,18 @@ def test_file_write():
     read_file.close()
 
 
+def test_file_write_span():
+    var content: String = "The quick brown fox jumps over the lazy dog"
+    var TEMP_FILE = Path(TEMP_FILE_DIR) / "test_file_write_span"
+    var f = open(TEMP_FILE, "w")
+    f.write(content.as_bytes_slice())
+    f.close()
+
+    var read_file = open(TEMP_FILE, "r")
```

Comment:
**Suggestion** Can we use a context manager here, e.g.

```
with open(TEMP_FILE, "r") as f:
  ...
```

Reply:
done.
I fixed a few other tests to use `with`.
note in `test_file_get_raw_fd`
```
    # since JIT and build give different file descriptors, we test by checking
    # if we printed to the right file.
    var f1 = open(Path(TEMP_FILE_DIR) / "test_file_dummy_1", "rw")
    var f2 = open(Path(TEMP_FILE_DIR) / "test_file_dummy_2", "rw")
    var f3 = open(Path(TEMP_FILE_DIR) / "test_file_dummy_2", "rw")
```
Currently Mojo does not support several statements in one `with`, e.g. `with xxx, yyy:`. This is supported in Python, the compiler team might want to support that.